### ISSUE TITLE
S4226 should not raise for generated class

### ISSUE THREAD

**Issue by lg2de** [SonarSource/sonar-dotnet#5457]
### Description

The Rule S4226 (Extensions should be in separate namespaces) notes in the description: "It makes little sense to create an extension method when it is possible to just add that method to the class itself."

If the base class is a generated class (e.g. using _xsd.exe_ or _xsd2code_) it is **not** "possible to just add that method to the class itself".

So, I ask to add exception for generated classes, e.g. annotated with `System.CodeDom.Compiler.GeneratedCodeAttribute`.

### Repro steps

```csharp
[System.CodeDom.Compiler.GeneratedCodeAttribute("xsd", "4.8.3928.0")]
[System.SerializableAttribute()]
[System.Diagnostics.DebuggerStepThroughAttribute()]
[System.ComponentModel.DesignerCategoryAttribute("code")]
public abstract partial class ReplyWithErrorListBase {
    
    private ReplyErrorListError[] errorListField;
    
    /// <remarks/>
    [System.Xml.Serialization.XmlArrayAttribute(Form=System.Xml.Schema.XmlSchemaForm.Unqualified)]
    [System.Xml.Serialization.XmlArrayItemAttribute("Error", Form=System.Xml.Schema.XmlSchemaForm.Unqualified, IsNullable=false)]
    public ReplyErrorListError[] ErrorList {
        get {
            return this.errorListField;
        }
        set {
            this.errorListField = value;
        }
    }
}

internal static class ReplyErrorExtensions
{
    public static void SetSyncLaterError(this ReplyWithErrorListBase reply) // SHOULD NOT RAISE!
    {
        reply.ErrorList = new[]
            { new ReplyErrorListError { Code = 1, Text = "foo" } };
    }
}
```

### Expected behavior

S4226 should not raise.

### Actual behavior

Warning: "Either move this extension to another namespace or move the method inside the class itself."

### Known workarounds

N/A

### Related information

* C#/VB.NET Plugins version: 8.26
* Visual Studio version: 2022
* MSBuild / dotnet version: net472
* Operating System: Windows 10
